### PR TITLE
M3-3783: Replace react-async-script-loader package w/ react-async-script

### PR DESCRIPTION
**Pull request by DevDW**
## Description
Replace [react-async-script-loader](https://www.npmjs.com/package/react-async-script-loader), which hasn't been updated in several years, with [react-async-script](https://www.npmjs.com/package/react-async-script).

The only place this was being used was the Payment drawer. I don't believe that `react-async-script` passes props as `react-async-script-loader` did, so this will require some adjusting to keep working as intended.

## Type of Change
- Non breaking change ('update', 'change')

## To-Do
- [x] Further tweaking and testing with `react-async-script` to ensure this is actually working

### CODE REVIEW

**Comment by DevDW** [comment]
https://github.com/linode/manager/blob/97801834f1eb5713445bba7b1de4ea22d5406903/packages/manager/src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/PaymentDrawer.tsx#L74

When you test, if you change the `paypalScriptSrc()` bit to some other random URL, the button should stop showing up (because `isPaypalScriptLoaded` should not get set to true in that case). So I believe this should be working as intended now, please let me know if it is not

**Review by Jskobos — APPROVED** [review]
Looks good. 

This is pre-existing, but maybe we should hide the full paypal section or show a loading/error state. If you block requests to *paypal*:

Screen Shot 2020-11-19 at 4 53 19 PM

**Comment by DevDW** [comment]
@Jskobos I was hoping to find a way to access the async wrapper's error state to make things simple, but the recent changes are in lieu of that... alternatively I can just hide the whole PayPal section if `isPayPalInitialized` is false.

File: packages/manager/src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/PaymentDrawer.tsx [hunk 1]
```diff
@@ -67,6 +67,11 @@ export const PaymentDrawer: React.FC<CombinedProps> = props => {
 
   const [creditCardKey, setCreditCardKey] = React.useState<string>(v4());
   const [payPalKey, setPayPalKey] = React.useState<string>(v4());
+  const [isPaypalScriptLoaded, setIsPaypalScriptLoaded] = React.useState<
+    boolean
+  >(false);
+
+  const AsyncPaypal = makeAsyncScriptLoader(paypalScriptSrc())(PayPal);
```
Comment (johnwcallahan) [inline packages/manager/src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/PaymentDrawer.tsx:74]: To me it seems this HOC shouldn't be called within the function body itself.

File: packages/manager/src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/Paypal.tsx [hunk 2]
```diff
@@ -137,10 +138,23 @@ export const PayPalDisplay: React.FC<CombinedProps> = props => {
...
       });
+      setErrorLoadingPaypalScript(false);
       setShouldRenderButton(true);
     }
   }, [isScriptLoaded, shouldRenderButton]);
 
+  React.useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    if ('paypal' in window === false) {
```
Comment (johnwcallahan) [inline packages/manager/src/features/Billing/BillingPanels/BillingSummary/PaymentDrawer/Paypal.tsx:149]: I don't think this is working as intended. The check for `'paypal' in window` happens _before_ the timeout is set, but it seems like the intention is to wait 2s, and if the script hasn't loaded by then, show an error. That's not what this code is doing though.

**Review by johnwcallahan — APPROVED** [review]

**Pull request merged**